pkgs/top-level/all-packages.nix

```diff
@@ -14259,6 +14259,7 @@ with pkgs;
       /**/ if platform.isDarwin then 11
       else if platform.isFreeBSD then 7
       else if platform.isAndroid then 12
+      else if platform.isAarch32 then 7
```

Comment:
This would affect armv7l too, as you noted in the commit message.

Can you provide details about reproducing issues you faced? I think it would be nice not to revert to llvm_7 on armv7l unless needed.

Reply:
Just try building `pkgsCross.raspberryPi.llvmPackages.compiler-rt-libc`.
It fails with some errors related to instructions not being available in thumb mode.

I don't know that's the armv7 equivalent be, `pkgsCross.armv7l-hf-multiplatform`?

Reply:
Yes it's `armv7l-hf-multiplatform`. The "example" platform names under `pkgsCross` are a proper mess...

Reply:
Another note: it might be nice to add a comment stating the change is for cross-compilation, that native compilation was untested.

Reply:
Yes, armv7 doesn't seem to be affected.

Reply:
Done.

Reply:
I looked into trying to get more recent versions of compiler-rt to build while trying to solve this issue a while ago. For reference, the error message is:
```
/build/compiler-rt-src-14.0.6/compiler-rt/lib/builtins/arm/sync_fetch_and_add_4.S: Assembler messages:
/build/compiler-rt-src-14.0.6/compiler-rt/lib/builtins/arm/sync_fetch_and_add_4.S:19: Error: selected processor does not support `dmb' in Thumb mode
/build/compiler-rt-src-14.0.6/compiler-rt/lib/builtins/arm/sync_fetch_and_add_4.S:19: Error: selected processor does not support `ldrex r0,[r12]' in Thumb mode
/build/compiler-rt-src-14.0.6/compiler-rt/lib/builtins/arm/sync_fetch_and_add_4.S:19: Error: cannot honor width suffix -- `add r2,r0,r1'
/build/compiler-rt-src-14.0.6/compiler-rt/lib/builtins/arm/sync_fetch_and_add_4.S:19: Error: selected processor does not support `strex r3,r2,[r12]' in Thumb mode
/build/compiler-rt-src-14.0.6/compiler-rt/lib/builtins/arm/sync_fetch_and_add_4.S:19: Error: selected processor does not support `dmb' in Thumb mode
```
There are a few patches included in LLVM 15 that partially fix the issue: https://reviews.llvm.org/D104183, https://reviews.llvm.org/D99282. Unfortunately, it still doesn't fix the [usage of `ldrexd` and `strexd`](https://github.com/llvm/llvm-project/blob/7c0362b1a1b59a5b7d164debcea7b3c6c6948542/compiler-rt/lib/builtins/arm/sync-ops.h#L48), which is not supported in base ARMv6.

It seemed a little odd that only a partial fix was included, but the person who submitted the patches was targeting a Raspberry Pi 0, which has a ARM1176JZF-S CPU. This CPU actually implements ARMv6k, which supports `ldrexd`/`strexd`. If they configured their toolchain to target that specific processor, they wouldn't run into an error when using those instructions.

Reply:
Also, I suspect something like this but with `__ARM_ARCH_6__` is needed to fix the issue: https://github.com/llvm/llvm-project/commit/609e2e6a6e001bcbe3da469a0c117fa570ca71aa